set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java

```diff
@@ -431,5 +431,28 @@ public void removeAll(List<PersistentUserSet> userSets) {
 	public PersistentUserSet update(PersistentUserSet userSet) throws  UserSetValidationException {
 		return store(userSet);
 	}
+	
+	/**
+	 * Getting the ids of the duplicate sets.
+	 */
+	public List<String> getDuplicateUserSetsIds(UserSet userSet) {
+	  
+	     if(userSet.getSubject()==null || userSet.getSubject().size()==0) return null;
```

Comment:
this check must be performed only for EntityBestItems type, no db request should be performed for other types

Reply:
Improved.